int32 NONE=0
int32 ZONE=1
int32 TIMER=2
int32 type